draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -835,75 +869,75 @@ Summary of quantities:
 | r | Order of G. | This number MUST be prime.  |
 | h | Cofactor, h >= 1. | An integer satisfying n = h * r.  |
 
 ## Terminology
-In the following, we categorize the terminology for encoding bitstrings to points
-on elliptic curves.
+
+In this section, we define terms used in the rest of this document.
 
 ### Encoding {#term-encoding}
 
-In practice, the input of a given cryptographic algorithm will be a bitstring of
-arbitrary length, denoted {0, 1}^\*. Hence, a concern for virtually all protocols
-involving elliptic curves is how to convert this input into a point on the curve.
-The general term "encoding" refers to the process of producing an elliptic curve
-point given as input a bitstring. In some protocols, the original message may
-also be recovered through a decoding procedure. An encoding may be deterministic
-or probabilistic, although the latter is a potential source of plaintext
-information leakage in face of side-channel attacks.
-
-Suppose that the input of the encoding function is a bitstring of fixed-length
-L. Comparing sizes of the sets, 2^L and n, an encoding function cannot be both
-deterministic and bijective. Alternatively, an injective encoding from {0, 1}^L
-to E can be used, such that L < log2(n) - 1, which is a bijection over a subset
-of points on E. This ensures that encoded plaintext messages can be recovered.
-
-In practice, encodings are commonly injective and invertible. Injective
-encodings map inputs to a subset of points on the curve. Invertible encodings
-allow computation of input bitstrings given a point on the curve.
+Encoding is the process of computing an elliptic curve point from an input bit string.
+An encoding may be deterministic or probabilistic.
+Deterministic procedures are generally preferred for security,
+because probabilistic encodings can leak information through side channels.
+
+In general, the set of points output by a deterministic encoding procedure
+may not be the full set of points on an elliptic curve E
+(i.e., it may not be surjective).
+Moreover, an encoding procedure may map distinct inputs to the same output
+(i.e., it may not be injective).
+For example, suppose that the input of an encoding is a bit string of fixed length L.
+If 2^L (the number of possible inputs) is not equal to n (the number of points
+on the curve) then such an encoding cannot be deterministic, injective, and surjective.
+If 2^L < n, however, such an encoding may be bijective (i.e., both injective and surjective)
+over a subset of the points on E.
+
+Encodings may be invertible, meaning that there is an efficient method of recovering
+a bit string msg from any point P produced by the encoding, such that applying the
+encoding to x yields P.
+This document does not discuss inversion algorithms; the encodings herein
+are designed to be difficult to invert, since this is a standard security
+requirement for cryptographic hash functions.
```

Comment:
This last sentence seems to imply that encodings have properties akin to cryptographic hash functions, which isn’t true, right? I’d trim everything after “does not discuss inversion algorithms;”